src/Utilities/Compiler/CodeMetrics/CodeAnalysisMetricData.cs

```diff
@@ -180,66 +179,65 @@ void appendChildren(string indent)
         }
 
         /// <summary>
-        /// Computes <see cref="CodeAnalysisMetricData"/> for the given <paramref name="compilation"/>.
+        /// Computes <see cref="CodeAnalysisMetricData"/> for the given <paramref name="context"/>.
         /// </summary>
-        public static Task<CodeAnalysisMetricData> ComputeAsync(Compilation compilation, CancellationToken cancellationToken)
+        public static Task<CodeAnalysisMetricData> ComputeAsync(CodeMetricsAnalysisContext context)
```

Comment:
Lets try to preserve the existing public API surface of code under  `CodeMetrics` folder. We insert the utilities assembly produced by the repo into VisualStudio, and this will cause a build break when we try to bump up the version number. Suggestion:

```csharp
[Obsolete("Use ComputeAsync(CodeMetricsAnalysisContext) instead")
public static Task<CodeAnalysisMetricData> ComputeAsync(Compilation compilation, CancellationToken cancellationToken)
{
   // Create new CodeMetricsAnalysisContext and invoke the new API
}

public static Task<CodeAnalysisMetricData> ComputeAsync(CodeMetricsAnalysisContext context)
{
```